yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -163,6 +163,23 @@ private[spark] class Client(
     appContext.setQueue(args.amQueue)
     appContext.setAMContainerSpec(containerContext)
     appContext.setApplicationType("SPARK")
+    sparkConf.getOption(CONF_SPARK_YARN_APPLICATION_TAGS)
+      .map(StringUtils.getTrimmedStringCollection(_))
+      .filter(!_.isEmpty())
+      .foreach { tagCollection =>
+        try {
+          val method = appContext.getClass().getMethod(
+            "setApplicationTags", classOf[java.util.Set[String]])
+          method.invoke(appContext, new java.util.HashSet[String](tagCollection))
+          logInfo("Applied setApplicationTags based on %s='%s'"
+            .format(CONF_SPARK_YARN_APPLICATION_TAGS, tagCollection))
+
+        } catch {
+          case e: NoSuchMethodException =>
+            logWarning("Ignoring conf %s='%s'; setApplicationTags missing in this version of YARN."
```

Comment:
I'd say "Ignoring spark.yarn.tags because this version of YARN does not support it"